### PR TITLE
Add javadoc for SqlSelectFlow - Array Queries

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
# Pull Request Merge Checklist
           
Issue identifier related to this PR: [680](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/680)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java [hunk 1]
```diff
@@ -144,8 +144,22 @@ public boolean awaitTermination(long timeout, TimeUnit unit)
 
     public interface TaskDecorator
     {
+        /**
+         * Receives an object an modifies its metadata.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java:148]: Do you think we should explain better what is this "object"?

This description doesn't help me understanding the modifications and what is this Runnable command object

File: presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java [hunk 2]
```diff
@@ -144,8 +144,22 @@ public boolean awaitTermination(long timeout, TimeUnit unit)
...
+         *
+         * @param command the object that will be modified
+         * @return the modified object
+         */
         Runnable decorate(Runnable command);
 
+        /**
+         * Receives an object an modifies its metadata.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java:158]: Do you think we should explain better what is this "object"?

This description doesn't help me understanding the modifications and what is this Callable task object

File: presto-main/src/main/java/io/prestosql/execution/TaskManager.java [hunk 3]
```diff
@@ -77,6 +77,16 @@
      */
     ListenableFuture<TaskStatus> getTaskStatus(TaskId taskId, TaskState currentState);
 
+    /**
+     * Changes the {@link io.prestosql.memory.MemoryPool} which query must be assigned.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/TaskManager.java]:
```suggestion
     * Changes the {@link io.prestosql.memory.MemoryPool} which a query must be assigned.
```

File: presto-main/src/main/java/io/prestosql/execution/TaskManager.java [hunk 4]
```diff
@@ -77,6 +77,16 @@
...
+     * <p>
+     * Presto has two memory pools:
+     * - General pool: It is used by default.
+     * - Reserved pool: It is used to run the heaviest query when
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/TaskManager.java:85]:
```suggestion
     * - Reserved pool: It is used to run the heaviest queries when
```

File: presto-main/src/main/java/io/prestosql/memory/LocalMemoryManager.java [hunk 5]
```diff
@@ -92,6 +92,16 @@ private void validateHeapHeadroom(NodeMemoryConfig config, long availableMemory)
         }
     }
 
+    /**
+     * Gets the information about the node's memory.
+     * <p>
+     * The returned information are:
+     * - Quantity of available processors
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/memory/LocalMemoryManager.java]:
```suggestion
     * - The number of available processors
```

File: presto-main/src/main/java/io/prestosql/memory/LocalMemoryManager.java [hunk 6]
```diff
@@ -92,6 +92,16 @@ private void validateHeapHeadroom(NodeMemoryConfig config, long availableMemory)
...
+     * - Quantity of total available memory
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/memory/LocalMemoryManager.java]:
```suggestion
     * - The total available memory
```

File: presto-main/src/main/java/io/prestosql/memory/MemoryInfo.java [hunk 7]
```diff
@@ -42,18 +42,33 @@ public MemoryInfo(
         this.availableProcessors = availableProcessors;
     }
 
+    /**
+     * Gets the quantity of available processors for the node.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/memory/MemoryInfo.java]:
```suggestion
     * Gets the number of available processors for the node.
```

File: presto-main/src/main/java/io/prestosql/memory/MemoryInfo.java [hunk 8]
```diff
@@ -42,18 +42,33 @@ public MemoryInfo(
...
+     *
+     * @return the quantity of available processors for the node
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/memory/MemoryInfo.java]:
```suggestion
     * @return the number of available processors for the node
```

File: presto-main/src/main/java/io/prestosql/memory/MemoryPoolAssignmentsRequest.java [hunk 9]
```diff
@@ -42,6 +42,14 @@ public String getCoordinatorId()
         return coordinatorId;
     }
 
+    /**
+     * Gets the version of the request.
+     * <p>
+     * It is used to ensure that consistency if requests
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/memory/MemoryPoolAssignmentsRequest.java]:
```suggestion
     * It is used to ensure the consistency if requests
```

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/io/prestosql/execution/TaskManager.java (hunk 4 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/execution/TaskManager.java:85]: The reserved pool just run one query per time, i think just keep in singular

File: presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java (hunk 1 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java:148]: explanation added

File: presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java (hunk 2 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/dispatcher/DecoratingListeningExecutorService.java:158]: explanation added
